src/cli-util.js

```diff
@@ -0,0 +1,144 @@
+"use strict";
+
+const path = require("path");
+const dashify = require("dashify");
+
+const resolver = require("./resolve-config");
+
+function getOptions(argv) {
+  return {
+    cursorOffset: getIntOption(argv, "cursor-offset"),
+    rangeStart: getIntOption(argv, "range-start"),
+    rangeEnd: getIntOption(argv, "range-end"),
+    useTabs: argv["use-tabs"],
+    semi: argv["semi"],
+    printWidth: getIntOption(argv, "print-width"),
+    tabWidth: getIntOption(argv, "tab-width"),
+    bracketSpacing: argv["bracket-spacing"],
+    singleQuote: argv["single-quote"],
+    jsxBracketSameLine: argv["jsx-bracket-same-line"],
+    filepath: argv["stdin-filepath"],
+    trailingComma: getTrailingComma(argv),
+    parser: getParserOption(argv)
+  };
+}
+
+function getParserOption(argv) {
+  const value = argv.parser;
+
+  if (value === undefined) {
+    return value;
+  }
+
+  // For backward compatibility. Deprecated in 0.0.10
+  if (argv["flow-parser"]) {
+    console.warn("`--flow-parser` is deprecated. Use `--parser flow` instead.");
+    return "flow";
+  }
+
+  return value;
+}
+
+function getIntOption(argv, optionName) {
+  const value = argv[optionName];
+
+  if (value === undefined) {
+    return value;
+  }
+
+  if (/^\d+$/.test(value)) {
+    return Number(value);
+  }
+
+  console.error(
```

Comment:
We should make this `throw` rather than log and exit. `getTrailingComma` already does this.

Reply:
Same thought, but there are two kind of exit code, `1` and `2`, I'm not sure if it'll affect something.

Reply:
This undocumented and seemingly arbitrary `1` vs `2` status code thing is sad. Part of me wants to just use `1` everywhere, part of me is afraid of unexpectedly breaking someone’s build.